apps/server/src/modules/tool/external-tool/service/datasheet-pdf.service.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { externalToolDatasheetTemplateDataFactory, externalToolFactory, userDoFactory } from '@shared/testing';
+import { UserDO } from '@shared/domain/domainobject';
+import { DatasheetPdfService } from './datasheet-pdf.service';
+import { ExternalTool, ExternalToolDatasheetTemplateData } from '../domain';
+import { CustomParameter } from '../../common/domain';
+
+describe(DatasheetPdfService.name, () => {
+	let module: TestingModule;
+	let service: DatasheetPdfService;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [DatasheetPdfService],
+		}).compile();
+
+		service = module.get(DatasheetPdfService);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	afterEach(() => {
+		jest.resetAllMocks();
+	});
+
+	describe('generatePdf', () => {
+		describe('when no error occurs', () => {
+			const setup = () => {
+				const user: UserDO = userDoFactory.buildWithId();
+
+				const externalTool: ExternalTool = externalToolFactory.withCustomParameters(1).build();
+				const params = externalTool.parameters as CustomParameter[];
+				const datasheetData: ExternalToolDatasheetTemplateData = externalToolDatasheetTemplateDataFactory
+					.withParameters(1, { name: params[0].name })
+					.build({
+						toolName: externalTool.name,
+						instance: 'dBildungscloud',
+						creatorName: `${user.firstName} ${user.lastName}`,
+					});
+
+				return { datasheetData };
+			};
+
+			it('should return Buffer', async () => {
+				const { datasheetData } = setup();
+
+				const result = await service.generatePdf(datasheetData);
+
+				expect(result).toEqual(expect.any(Uint8Array));
```

Comment:
why do you expect a Uint8Array when the function returns a buffer?

Reply:
because expect.any(Buffer) did not work and the the debugger showed the type as Uint8Array 🤷‍♂️